diff -urN kget-4.2.87.old/cmake/modules/FindLibKNotificationItem-1.cmake kget-4.2.87/cmake/modules/FindLibKNotificationItem-1.cmake
--- kget-4.2.87.old/cmake/modules/FindLibKNotificationItem-1.cmake	2009-05-19 16:06:49.000000000 +0400
+++ kget-4.2.87/cmake/modules/FindLibKNotificationItem-1.cmake	1970-01-01 03:00:00.000000000 +0300
@@ -1,37 +0,0 @@
-# - Find libknotification-1 library
-# Find the libknotification-1. This is an experimental library which is not 
-# supposed to stay source- or binary compatible, it may even go away in the future again.
-#
-# This module defines
-#  LIBKNOTIFICATIONITEM-1_FOUND - whether the libkonitification-1 library was found
-#  LIBKNOTIFICATIONITEM-1_LIBRARIES - the libknotification-1 library
-#  LIBKNOTIFICATIONITEM-1_INCLUDE_DIR - the include path of the libknotification-1 library
-
-
-
-find_library (LIBKNOTIFICATIONITEM-1_LIBRARY
-  NAMES
-  knotificationitem-1
-  HINTS
-  ${LIB_INSTALL_DIR}
-  ${KDE4_LIB_DIR}
-)
-
-set(LIBKNOTIFICATIONITEM-1_LIBRARIES  ${LIBKNOTIFICATIONITEM-1_LIBRARY})
-
-find_path (LIBKNOTIFICATIONITEM-1_INCLUDE_DIR
-  NAMES
-  knotificationitem.h
-  PATH_SUFFIXES
-  knotificationitem-1
-  HINTS
-  ${INCLUDE_INSTALL_DIR}
-  ${KDE4_INCLUDE_DIR}
-)
-
-
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(LibKNotificationItem-1  DEFAULT_MSG  LIBKNOTIFICATIONITEM-1_LIBRARY LIBKNOTIFICATIONITEM-1_INCLUDE_DIR)
-
-mark_as_advanced(LIBKNOTIFICATIONITEM-1_INCLUDE_DIR LIBKNOTIFICATIONITEM-1_LIBRARY)
diff -urN kget-4.2.87.old/cmake/modules/FindLibKNotificationItem-1_New.cmake kget-4.2.87/cmake/modules/FindLibKNotificationItem-1_New.cmake
--- kget-4.2.87.old/cmake/modules/FindLibKNotificationItem-1_New.cmake	1970-01-01 03:00:00.000000000 +0300
+++ kget-4.2.87/cmake/modules/FindLibKNotificationItem-1_New.cmake	2009-05-19 16:06:49.000000000 +0400
@@ -0,0 +1,37 @@
+# - Find libknotification-1 library
+# Find the libknotification-1. This is an experimental library which is not 
+# supposed to stay source- or binary compatible, it may even go away in the future again.
+#
+# This module defines
+#  LIBKNOTIFICATIONITEM-1_FOUND - whether the libkonitification-1 library was found
+#  LIBKNOTIFICATIONITEM-1_LIBRARIES - the libknotification-1 library
+#  LIBKNOTIFICATIONITEM-1_INCLUDE_DIR - the include path of the libknotification-1 library
+
+
+
+find_library (LIBKNOTIFICATIONITEM-1_LIBRARY
+  NAMES
+  knotificationitem-1
+  HINTS
+  ${LIB_INSTALL_DIR}
+  ${KDE4_LIB_DIR}
+)
+
+set(LIBKNOTIFICATIONITEM-1_LIBRARIES  ${LIBKNOTIFICATIONITEM-1_LIBRARY})
+
+find_path (LIBKNOTIFICATIONITEM-1_INCLUDE_DIR
+  NAMES
+  knotificationitem.h
+  PATH_SUFFIXES
+  knotificationitem-1
+  HINTS
+  ${INCLUDE_INSTALL_DIR}
+  ${KDE4_INCLUDE_DIR}
+)
+
+
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(LibKNotificationItem-1  DEFAULT_MSG  LIBKNOTIFICATIONITEM-1_LIBRARY LIBKNOTIFICATIONITEM-1_INCLUDE_DIR)
+
+mark_as_advanced(LIBKNOTIFICATIONITEM-1_INCLUDE_DIR LIBKNOTIFICATIONITEM-1_LIBRARY)
diff -urN kget-4.2.87.old/CMakeLists.txt kget-4.2.87/CMakeLists.txt
--- kget-4.2.87.old/CMakeLists.txt	2009-05-21 20:15:55.000000000 +0400
+++ kget-4.2.87/CMakeLists.txt	2009-05-21 20:16:29.932032646 +0400
@@ -46,7 +46,7 @@
 macro_optional_find_package(JPEG)
 macro_log_feature(JPEG_FOUND "libjpeg" "Development library for JPEG formatted image files" "http://www.ijg.org/" FALSE "" "Provides JPEG images in Krdc.")
 
-macro_optional_find_package(LibKNotificationItem-1)
+macro_optional_find_package(LibKNotificationItem-1_New)
 macro_log_feature(LIBKNOTIFICATIONITEM-1_FOUND "libknotificationitem" "The KDE4 implementation of the experimental next-gen systray protocol" "" FALSE "" "Provides enhanced systray in KGet.")
 
 macro_optional_find_package(Soprano)
