--- kopete/cmake/modules/FindLibgadu.cmake	2009-06-17 22:08:12.000000000 +0200
+++ kopete/cmake/modules/FindLibgadu.cmake-new	2009-07-04 16:49:57.586710508 +0200
@@ -23,32 +23,18 @@
         find_package (PkgConfig)
         pkg_check_modules (PC_LIBGADU libgadu)
 
-        # If pkgconfig found libgadu, get the full path to the library using find_library()
-        # but only in the path reported by pkgconfig.
-        # Otherwise do a normal search.
-        if (PC_LIBGADU_FOUND)
-
-            set (LIBGADU_DEFINITIONS ${PC_LIBGADU_CFLAGS})
-
-            if (PC_LIBGADU_INCLUDE_DIRS)
-                set (LIBGADU_INCLUDE_DIR ${PC_LIBGADU_INCLUDE_DIRS})
-            else (PC_LIBGADU_INCLUDE_DIRS)
-                find_path (LIBGADU_INCLUDE_DIR libgadu.h PATH_SUFFIXES libgadu)
-            endif (PC_LIBGADU_INCLUDE_DIRS)
+            find_path (LIBGADU_INCLUDE_DIR libgadu.h
+		PATHS
+	    	/usr/include	
+		)
 
             find_library (LIBGADU_LIBRARIES NAMES gadu
                 PATHS
-                ${PC_LIBGADU_LIBDIR}
+		/usr/lib64
+		/usr/lib
                 NO_DEFAULT_PATH
             )
 
-        else (PC_LIBGADU_FOUND)
-
-            find_library (LIBGADU_LIBRARIES NAMES gadu)
-            find_path (LIBGADU_INCLUDE_DIR libgadu.h PATH_SUFFIXES libgadu)
-
-        endif (PC_LIBGADU_FOUND)
-
     else (NOT WIN32)
 
         find_library_with_debug (LIBGADU_LIBRARIES

