diff -Nru kdenetwork-4.1.61/krdc/CMakeLists.txt kdenetwork-4.1.61.new/krdc/CMakeLists.txt
--- kdenetwork-4.1.61/krdc/CMakeLists.txt     2008-08-07 17:22:17.000000000 +0200
+++ kdenetwork-4.1.61.new/krdc/CMakeLists.txt 2008-08-09 23:28:16.023378750 +0200
@@ -112,6 +112,6 @@
 install(FILES pointcursor.png pointcursormask.png DESTINATION ${DATA_INSTALL_DIR}/krdc/pics)
 
 if(LIBNXCL_FOUND)
-   install(FILES nx/default.dsa.key DESTINATION ${DATA_INSTALL_DIR}/krdc)
+   #install(FILES nx/default.dsa.key DESTINATION ${DATA_INSTALL_DIR}/krdc)
 endif(LIBNXCL_FOUND)
 
diff -ruN kdenetwork-4.1.64./krdc/nx/nxclientthread.cpp kdenetwork-4.1.64/krdc/nx/nxclientthread.cpp
--- kdenetwork-4.1.64./krdc/nx/nxclientthread.cpp	2008-08-31 15:43:51.000000000 +0200
+++ kdenetwork-4.1.64/krdc/nx/nxclientthread.cpp	2008-08-31 15:48:51.858519360 +0200
@@ -187,13 +187,6 @@
             p->probeProcess();
         }
 
-	if (!this->m_xid) {
-	    this->m_xid = m_client.getXID();
-
-	    if (this->m_xid) 
-	        emit hasXid(this->m_xid);
-        }
-
         usleep (1000);
     }
 }
