Index: krfb/krfb.notifyrc
===================================================================
--- krfb/krfb.notifyrc	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ krfb/krfb.notifyrc	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -53,8 +53,8 @@
 Comment[tg]=Истифодаи Муштараки Мизи Корӣ
 Comment[tr]=Masaüstü Paylaşımı
 Comment[uk]=Спільні стільниці
+Comment[xh]=Ulwahlulelano lwe Desktop
 Comment[x-test]=xxDesktop Sharingxx
-Comment[xh]=Ulwahlulelano lwe Desktop
 Comment[zh_CN]=桌面共享
 Comment[zh_HK]=桌面分享
 Comment[zh_TW]=桌面分享
@@ -156,8 +156,8 @@
 Comment[tg]=Корванд пайвастшавиро қабул мекунад
 Comment[tr]=Kullanıcı bağlantıyı kabul etti
 Comment[uk]=Користувач приймає з’єднання
+Comment[xh]=Umsebenzisi wamkela uxhulumaniso
 Comment[x-test]=xxUser accepts connectionxx
-Comment[xh]=Umsebenzisi wamkela uxhulumaniso
 Comment[zh_CN]=用户接受连接
 Comment[zh_HK]=用戶接受連線
 Comment[zh_TW]=使用者接受連線
@@ -260,8 +260,8 @@
 Comment[tg]=Корванд пайвастшавиро рад мекунад
 Comment[tr]=Kullanıcı bağlantıyı reddetti
 Comment[uk]=Користувач не приймає з’єднання
+Comment[xh]=Umsebenzisi wala uxhulumaniso
 Comment[x-test]=xxUser refuses connectionxx
-Comment[xh]=Umsebenzisi wala uxhulumaniso
 Comment[zh_CN]=用户拒绝连接
 Comment[zh_HK]=用戶拒絕連線
 Comment[zh_TW]=使用者拒絕連線使用者
@@ -292,6 +292,7 @@
 Name[ko]=연결이 닫힘
 Name[lt]=Ryšys baigtas
 Name[lv]=Savienojums slēgts
+Name[mai]=संबंधन बन्न भ' गेल
 Name[ml]=ബന്ധം അടച്ചു
 Name[nb]=Forbindelsen lukket
 Name[nds]=Afkoppelt
@@ -367,8 +368,8 @@
 Comment[uk]=З'єднання закрито
 Comment[uz]=Aloqa uzildi
 Comment[uz@cyrillic]=Алоқа узилди
+Comment[xh]=Uxhulumaniso luvaliwe
 Comment[x-test]=xxConnection closedxx
-Comment[xh]=Uxhulumaniso luvaliwe
 Comment[zh_CN]=连接关闭
 Comment[zh_HK]=連線已關閉
 Comment[zh_TW]=連線已關閉
@@ -450,6 +451,7 @@
 Comment[ko]=잘못된 암호
 Comment[lt]=Neteisingas slaptažodis
 Comment[lv]=Parole nav derīga
+Comment[mai]=अवैध कूटशब्द
 Comment[mk]=Невалидна лозинка
 Comment[ml]=അസാധുവായ അടയാളവാക്ക്
 Comment[ms]=Kata laluan tidak sah
@@ -476,8 +478,8 @@
 Comment[uz]=Maxfiy soʻz haqiqiy emas
 Comment[uz@cyrillic]=Махфий сўз ҳақиқий эмас
 Comment[wa]=Sicret nén valide
+Comment[xh]=Igama lokugqitha elingasebenziyo
 Comment[x-test]=xxInvalid passwordxx
-Comment[xh]=Igama lokugqitha elingasebenziyo
 Comment[zh_CN]=无效密码
 Comment[zh_HK]=無效的密碼
 Comment[zh_TW]=不正確的密碼
@@ -579,8 +581,8 @@
 Comment[tg]=Корванди дурдаст гузарвожаи нодурустро фиристод. Пайвастшавӣ манъ шудааст.
 Comment[tr]=Davet edilenden gönderilmiş geçersiz parola. Bağlantı rededildi.
 Comment[uk]=Запрошений учасник надіслав некоректний пароль. У з’єднанні відмовлено.
+Comment[xh]=Umhlangano omenyiweyo uthumele igama lokugqitha elisebenzayo. Uxhulumano lwa liwe.
 Comment[x-test]=xxThe invited party sent an invalid password. Connection refused.xx
-Comment[xh]=Umhlangano omenyiweyo uthumele igama lokugqitha elisebenzayo. Uxhulumano lwa liwe.
 Comment[zh_CN]=受邀请方发送的密码不对。连接被拒绝。
 Comment[zh_HK]=被邀請的一方送出無效的密碼。已拒絕連線。
 Comment[zh_TW]=邀請的人送出了不合法的密碼邀請。連線已拒絕。
@@ -682,8 +684,8 @@
 Comment[tg]=Пайвастшавӣ дархоста шудааст, корванд бояд қабул кунад
 Comment[tr]=Bağlantı isteği, kullanıcı kabul etmeli
 Comment[uk]=Отримано запит на з’єднання, користувач має його прийняти
+Comment[xh]=Uxhulumaniso luceliwe, umsebenzisi kufanele amkele
 Comment[x-test]=xxConnection requested, user must acceptxx
-Comment[xh]=Uxhulumaniso luceliwe, umsebenzisi kufanele amkele
 Comment[zh_CN]=连接已请求，用户必须接受
 Comment[zh_HK]=已請求連線，用戶必須接受
 Comment[zh_TW]=連線已要求，必須等使用者接受
@@ -786,8 +788,8 @@
 Comment[tg]=Пайвастшавии нав ба таври худкор барпо мегардад
 Comment[tr]=Yeni bağlantı otomatik olarak kuruldu
 Comment[uk]=Автоматично встановлено нове з’єднання
+Comment[xh]=Uxhulumaniso olutsha lufunyenwe ngokuzenzekelayo
 Comment[x-test]=xxNew connection automatically establishedxx
-Comment[xh]=Uxhulumaniso olutsha lufunyenwe ngokuzenzekelayo
 Comment[zh_CN]=自动建立新连接
 Comment[zh_HK]=已自動建立新連線
 Comment[zh_TW]=新連線自動建立
@@ -893,8 +895,8 @@
 Comment[uk]=Зайнято, у з’єднанні відмовлено
 Comment[uz]=Band, aloqa rad etildi
 Comment[uz@cyrillic]=Банд, алоқа рад этилди
+Comment[xh]=Uxhulumaniso, olu xakekileyo lwaliwe
 Comment[x-test]=xxBusy, connection refusedxx
-Comment[xh]=Uxhulumaniso, olu xakekileyo lwaliwe
 Comment[zh_CN]=对方处于忙碌状态，连接被拒绝
 Comment[zh_HK]=忙碌，已拒絕連線
 Comment[zh_TW]=忙碌，連線被拒
@@ -997,8 +999,8 @@
 Comment[tg]=Пайвастшавии ғайричашмдош қабул гардид, кандашавӣ
 Comment[tr]=Babul edilmiş beklenmedik bağlantı, durdur
 Comment[uk]=Отримано з’єднання, яке не очікувалось, припиняється
+Comment[xh]=Ufumene uxhulumaniso olungalindelekanga, lahla
 Comment[x-test]=xxReceived unexpected connection, abortxx
-Comment[xh]=Ufumene uxhulumaniso olungalindelekanga, lahla
 Comment[zh_CN]=收到意外连接，已中止
 Comment[zh_HK]=接收到非預期的連線，中止
 Comment[zh_TW]=已接收到未知的連線，中止。
Index: krfb/kcm_krfb/kcmkrfb.desktop
===================================================================
--- krfb/kcm_krfb/kcmkrfb.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ krfb/kcm_krfb/kcmkrfb.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -69,8 +69,8 @@
 Name[tr]=Masaüstü Paylaşımı
 Name[uk]=Спільні стільниці
 Name[vi]=Chia sẻ màn hình nền
+Name[xh]=Ukwehlulelana kwe Desktop
 Name[x-test]=xxDesktop Sharingxx
-Name[xh]=Ukwehlulelana kwe Desktop
 Name[zh_CN]=桌面共享
 Name[zh_HK]=桌面分享
 Name[zh_TW]=桌面分享
@@ -134,8 +134,8 @@
 Comment[tr]=Masaüstü Paylaşımını Yapılandır
 Comment[uk]=Налаштувати спільні стільниці
 Comment[vi]=Cấu hình chia sẻ màn hình nền
+Comment[xh]=Qwalasela Ukusebenzisa ngokuhlangeneyo kwe Desktop 
 Comment[x-test]=xxConfigure Desktop Sharingxx
-Comment[xh]=Qwalasela Ukusebenzisa ngokuhlangeneyo kwe Desktop 
 Comment[zh_CN]=配置桌面共享
 Comment[zh_HK]=設定桌面分享
 Comment[zh_TW]=設定桌面分享
Index: krfb/kinetd/kinetd.desktop
===================================================================
--- krfb/kinetd/kinetd.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ krfb/kinetd/kinetd.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -130,8 +130,8 @@
 Comment[uz]=Talab qilinganda tarmoq xizmatlarini ishga tushuruvchi Internet demoni
 Comment[uz@cyrillic]=Талаб қилинганда тармоқ хизматларини ишга тушурувчи Интернет демони
 Comment[vi]=Một trình nền Mạng mà chạy các dịch vụ mạng theo nhu cầu
+Comment[xh]=Internet daemon eqala iinkonzo zomsebenzi wonatha xa zifunwa
 Comment[x-test]=xxAn Internet daemon that starts network services on demandxx
-Comment[xh]=Internet daemon eqala iinkonzo zomsebenzi wonatha xa zifunwa
 Comment[zh_CN]=按需启动网络服务的守护进程
 Comment[zh_HK]=自動依需求起動網絡服務的互聯網系統程式
 Comment[zh_TW]=依要求起動網路服務的 Internet 伺服程式
Index: krfb/kinetd/kinetd.notifyrc
===================================================================
--- krfb/kinetd/kinetd.notifyrc	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ krfb/kinetd/kinetd.notifyrc	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -58,8 +58,8 @@
 Name[ta]=உள்வரும் இணைப்புகள்
 Name[tg]=Пайвастшавии Воридшаванда
 Name[tr]=Gelen Bağlantı
+Name[xh]=Uxhumaniso Olungenayo
 Name[x-test]=xxIncomingConnectionxx
-Name[xh]=Uxhumaniso Olungenayo
 Name[zh_CN]=进入的连接
 Name[zh_HK]=進入連線
 Name[zh_TW]=進來的連線
@@ -114,8 +114,8 @@
 Comment[tg]=Пайвастшавии воридшаванда қабул гардид
 Comment[tr]=Gelen bağlantı alınıyor
 Comment[uk]=Отримано вхідне з’єднання
+Comment[xh]=Uxhulumano olungenayo olufunyenweyo
 Comment[x-test]=xxReceived incoming connectionxx
-Comment[xh]=Uxhulumano olungenayo olufunyenweyo
 Comment[zh_CN]=收到进入的连接
 Comment[zh_HK]=已接收的進入連線
 Comment[zh_TW]=接收到進來的連線
@@ -125,7 +125,7 @@
 Name=ProcessFailed
 Name[ar]=فشلت العملية
 Name[bn]=প্রসেস ব্যর্থ
-Name[ca]=Procés fallit
+Name[ca]=Procés erroni
 Name[cs]=Proces selhal
 Name[cy]=MethoddProses
 Name[da]=ProcesMislykkedes
@@ -169,8 +169,8 @@
 Name[ta]=செயல் தோல்வியுற்றது
 Name[tg]=Ҷараён бо Нокомӣ анҷомид
 Name[tr]=Başarısız İşlemler
+Name[xh]=Inkqubo Yahlulekile
 Name[x-test]=xxProcessFailedxx
-Name[xh]=Inkqubo Yahlulekile
 Name[zh_CN]=处理失败
 Name[zh_HK]=程序失敗
 Name[zh_TW]=行程失敗
@@ -225,8 +225,8 @@
 Comment[tg]=Барои даскории пайвастшавӣ ҷараён бозхонда нашуд
 Comment[tr]=Elle bağlantıda başarısız çağrı işlemleri
 Comment[uk]=Не вдалося викликати процес для з’єднання
+Comment[xh]=Ayikwazanga ukubiza inkqubo ezakuphatha uxhulumano
 Comment[x-test]=xxCould not call process to handle connectionxx
-Comment[xh]=Ayikwazanga ukubiza inkqubo ezakuphatha uxhulumano
 Comment[zh_CN]=无法调用进程处理连接
 Comment[zh_HK]=無法呼叫處理連線的程序
 Comment[zh_TW]=無法呼叫行程來處理此連線
Index: krfb/kinetd/kinetdmodule.desktop
===================================================================
--- krfb/kinetd/kinetdmodule.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ krfb/kinetd/kinetdmodule.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -60,8 +60,8 @@
 Name[tr]=KDED Modül Türü
 Name[uk]=Тип модулю KinetD
 Name[vi]=Kiểu môđun KInetD
+Name[xh]=Udidi Lomqongo womlinganiselo we KInetD
 Name[x-test]=xxKInetD Module Typexx
-Name[xh]=Udidi Lomqongo womlinganiselo we KInetD
 Name[zh_CN]=KInetD 模块类型
 Name[zh_HK]=KInetD 模組類型
 Name[zh_TW]=KInetD 模組類型
Index: kopete/plugins/nowlistening/kopete_nowlistening_config.desktop
===================================================================
--- kopete/plugins/nowlistening/kopete_nowlistening_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/nowlistening/kopete_nowlistening_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -40,6 +40,7 @@
 Name[ko]=지금 재생 중
 Name[lt]=Dabar klausau
 Name[lv]=Šobrīd atskaņo
+Name[mai]=आबि सुनि रहल अछि
 Name[mk]=Сега слуша
 Name[ml]=ഇപ്പോള്‍ ശ്രദ്ധിക്കുന്നു
 Name[nb]=Hører nå på
Index: kopete/plugins/nowlistening/kopete_nowlistening.desktop
===================================================================
--- kopete/plugins/nowlistening/kopete_nowlistening.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/nowlistening/kopete_nowlistening.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -45,6 +45,7 @@
 Name[ko]=지금 재생 중
 Name[lt]=Dabar klausau
 Name[lv]=Šobrīd atskaņo
+Name[mai]=आबि सुनि रहल अछि
 Name[mk]=Сега слуша
 Name[ml]=ഇപ്പോള്‍ ശ്രദ്ധിക്കുന്നു
 Name[nb]=Hører nå på
Index: kopete/plugins/alias/kopete_alias.desktop
===================================================================
--- kopete/plugins/alias/kopete_alias.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/alias/kopete_alias.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -37,6 +37,7 @@
 Name[ko]=별명
 Name[lt]=Kitas vardas
 Name[lv]=Īsceļi
+Name[mai]=उपनाम
 Name[mk]=Други имиња
 Name[ml]=ചെല്ലപ്പേര്
 Name[ne]=उपनाम
Index: kopete/plugins/alias/kopete_alias_config.desktop
===================================================================
--- kopete/plugins/alias/kopete_alias_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/alias/kopete_alias_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -32,6 +32,7 @@
 Name[ko]=별명
 Name[lt]=Kitas vardas
 Name[lv]=Īsceļi
+Name[mai]=उपनाम
 Name[mk]=Други имиња
 Name[ml]=ചെല്ലപ്പേര്
 Name[ne]=उपनाम
Index: kopete/plugins/translator/kopete_translator.desktop
===================================================================
--- kopete/plugins/translator/kopete_translator.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/translator/kopete_translator.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -46,6 +46,7 @@
 Name[ku]=Wergêr
 Name[lt]=Vertėjas
 Name[lv]=Tulkotājs
+Name[mai]=अनुवादक
 Name[mk]=Преведувач
 Name[ml]=പരിഭാഷകന്‍
 Name[nb]=Oversetter
Index: kopete/plugins/translator/kopete_translator_config.desktop
===================================================================
--- kopete/plugins/translator/kopete_translator_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/translator/kopete_translator_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -41,6 +41,7 @@
 Name[ku]=Wergêr
 Name[lt]=Vertėjas
 Name[lv]=Tulkotājs
+Name[mai]=अनुवादक
 Name[mk]=Преведувач
 Name[ml]=പരിഭാഷകന്‍
 Name[nb]=Oversetter
Index: kopete/plugins/statistics/kopete_statistics.desktop
===================================================================
--- kopete/plugins/statistics/kopete_statistics.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/statistics/kopete_statistics.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -46,6 +46,7 @@
 Name[ku]=Statistîk
 Name[lt]=Statistika
 Name[lv]=Statistika
+Name[mai]=सांख्यिकी
 Name[ml]=സ്ഥിതിവിവരം
 Name[nb]=Statistikk
 Name[nds]=Statistik
Index: kopete/plugins/addbookmarks/kopete_addbookmarks.desktop
===================================================================
--- kopete/plugins/addbookmarks/kopete_addbookmarks.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/addbookmarks/kopete_addbookmarks.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -33,6 +33,7 @@
 Name[ko]=책갈피
 Name[lt]=Žymelės
 Name[lv]=Grāmatzīmes
+Name[mai]=पुस्तकचिह्न
 Name[mk]=Обележувачи
 Name[ml]=ഓര്‍മ്മക്കുറിപ്പുകള്‍
 Name[nb]=Bokmerker
@@ -58,8 +59,8 @@
 Name[uz]=Xatchoʻplar
 Name[uz@cyrillic]=Хатчўплар
 Name[vi]=Dấu hiệu
+Name[xh]=Amanqaku encwadi
 Name[x-test]=xxBookmarksxx
-Name[xh]=Amanqaku encwadi
 Name[zh_CN]=书签
 Name[zh_HK]=書籤
 Name[zh_TW]=書籤
Index: kopete/plugins/addbookmarks/kopete_addbookmarks_config.desktop
===================================================================
--- kopete/plugins/addbookmarks/kopete_addbookmarks_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/addbookmarks/kopete_addbookmarks_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -33,6 +33,7 @@
 Name[ko]=책갈피
 Name[lt]=Žymelės
 Name[lv]=Grāmatzīmes
+Name[mai]=पुस्तकचिह्न
 Name[mk]=Обележувачи
 Name[ml]=ഓര്‍മ്മക്കുറിപ്പുകള്‍
 Name[nb]=Bokmerker
@@ -58,8 +59,8 @@
 Name[uz]=Xatchoʻplar
 Name[uz@cyrillic]=Хатчўплар
 Name[vi]=Dấu hiệu
+Name[xh]=Amanqaku encwadi
 Name[x-test]=xxBookmarksxx
-Name[xh]=Amanqaku encwadi
 Name[zh_CN]=书签
 Name[zh_HK]=書籤
 Name[zh_TW]=書籤
Index: kopete/plugins/history/kopete_history.desktop
===================================================================
--- kopete/plugins/history/kopete_history.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/history/kopete_history.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -47,6 +47,7 @@
 Name[ko]=과거 기록
 Name[lt]=Žurnalas
 Name[lv]=Vēsture
+Name[mai]=इतिहास
 Name[mk]=Историја
 Name[ml]=നാള്‍വഴി
 Name[nb]=Historie
@@ -75,8 +76,8 @@
 Name[uz@cyrillic]=Тарих
 Name[vi]=Lịch sử
 Name[wa]=Istwere
+Name[xh]=Imbali
 Name[x-test]=xxHistoryxx
-Name[xh]=Imbali
 Name[zh_CN]=历史
 Name[zh_HK]=歷程紀錄
 Name[zh_TW]=歷史
Index: kopete/plugins/history/kopete_history_config.desktop
===================================================================
--- kopete/plugins/history/kopete_history_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/history/kopete_history_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -42,6 +42,7 @@
 Name[ko]=과거 기록
 Name[lt]=Žurnalas
 Name[lv]=Vēsture
+Name[mai]=इतिहास
 Name[mk]=Историја
 Name[ml]=നാള്‍വഴി
 Name[nb]=Historie
@@ -70,8 +71,8 @@
 Name[uz@cyrillic]=Тарих
 Name[vi]=Lịch sử
 Name[wa]=Istwere
+Name[xh]=Imbali
 Name[x-test]=xxHistoryxx
-Name[xh]=Imbali
 Name[zh_CN]=历史
 Name[zh_HK]=歷程紀錄
 Name[zh_TW]=歷史
Index: kopete/plugins/highlight/kopete_highlight.desktop
===================================================================
--- kopete/plugins/highlight/kopete_highlight.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/highlight/kopete_highlight.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -46,6 +46,7 @@
 Name[ko]=강조
 Name[lt]=Paryškinti
 Name[lv]=Izcelta
+Name[mai]=उभारू
 Name[mk]=Осветлување
 Name[ml]=എടുത്തുകാട്ടുക
 Name[nb]=Marker
Index: kopete/plugins/highlight/kopete_highlight_config.desktop
===================================================================
--- kopete/plugins/highlight/kopete_highlight_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/highlight/kopete_highlight_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -40,6 +40,7 @@
 Name[ko]=강조
 Name[lt]=Paryškinti
 Name[lv]=Izcelta
+Name[mai]=उभारू
 Name[mk]=Осветлување
 Name[ml]=എടുത്തുകാട്ടുക
 Name[nb]=Marker
Index: kopete/plugins/privacy/kopete_privacy_config.desktop
===================================================================
--- kopete/plugins/privacy/kopete_privacy_config.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/privacy/kopete_privacy_config.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -32,6 +32,7 @@
 Name[ko]=사생활
 Name[lt]=Privatumas
 Name[lv]=Privātums
+Name[mai]=गुप्तता
 Name[ml]=സ്വകാര്യ‌ത
 Name[nb]=Personvern
 Name[nds]=Privaatrebeet
Index: kopete/plugins/privacy/kopete_privacy.desktop
===================================================================
--- kopete/plugins/privacy/kopete_privacy.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/plugins/privacy/kopete_privacy.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -38,6 +38,7 @@
 Name[ko]=사생활
 Name[lt]=Privatumas
 Name[lv]=Privātums
+Name[mai]=गुप्तता
 Name[ml]=സ്വകാര്യ‌ത
 Name[nb]=Personvern
 Name[nds]=Privaatrebeet
Index: kopete/protocols/yahoo/kopete_yahoo.desktop
===================================================================
--- kopete/protocols/yahoo/kopete_yahoo.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/protocols/yahoo/kopete_yahoo.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -21,6 +21,7 @@
 Name[hi]=याहू
 Name[hne]=याहू
 Name[km]=យ៉ាហ៊ូ
+Name[mai]=याहू
 Name[ml]=യാഹൂ
 Name[ne]=याहू
 Name[pa]=ਯਾਹੂ
Index: kopete/protocols/jabber/jabbercontact.cpp
===================================================================
--- kopete/protocols/jabber/jabbercontact.cpp	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/protocols/jabber/jabbercontact.cpp	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -1327,7 +1327,6 @@
 		if(!Kopete::AccountManager::self()->registerAccount( transport ))
 			return;
 		
-		transport->setIdentity( parentAccount->identity() );
 		transport->myself()->setOnlineStatus( status ); //push back the online status
 		return;
 	}
Index: kopete/protocols/jabber/jabberchatsession.cpp
===================================================================
--- kopete/protocols/jabber/jabberchatsession.cpp	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/protocols/jabber/jabberchatsession.cpp	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -146,7 +146,7 @@
 	Kopete::ContactPtrList chatMembers = members ();
 
 	// make sure we do have members in the chat
-	if ( !chatMembers.first () )
+	if ( chatMembers.isEmpty () || !chatMembers.first () )
 		return;
 
 	XMPP::Jid jid = static_cast<JabberBaseContact*>(chatMembers.first())->rosterItem().jid();
Index: kopete/kopete/kopete.notifyrc
===================================================================
--- kopete/kopete/kopete.notifyrc	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/kopete.notifyrc	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -69,6 +69,7 @@
 Name[ko]=그룹
 Name[lt]=Grupė
 Name[lv]=Grupa
+Name[mai]=समूह
 Name[ml]=കൂട്ടം
 Name[nb]=Gruppe
 Name[nds]=Koppel
@@ -157,6 +158,7 @@
 Name[ko]=대화 상대
 Name[lt]=Kontaktinis asmuo
 Name[lv]=Kontakts
+Name[mai]=संपर्क
 Name[ml]=സമ്പര്‍ക്കം
 Name[nb]=Kontakt
 Name[nds]=Kontakt
@@ -245,6 +247,7 @@
 Name[ko]=등급
 Name[lt]=Klasė
 Name[lv]=Klase
+Name[mai]=वर्ग
 Name[ml]=വിഭാഗം
 Name[nb]=Klasse
 Name[nds]=Klass
@@ -730,6 +733,7 @@
 Name[ko]=오프라인
 Name[lt]=Atsijungę
 Name[lv]=Nesaistē
+Name[mai]=आफलाइन
 Name[mk]=Не на линија
 Name[ml]=ഓഫ്‌ലൈന്‍‍
 Name[nb]=Frakoblet
@@ -959,6 +963,7 @@
 Name[ko]=강조
 Name[lt]=Paryškinti
 Name[lv]=Izcelta
+Name[mai]=उभारू
 Name[mk]=Осветлување
 Name[ml]=എടുത്തുകാട്ടുക
 Name[nb]=Marker
@@ -1584,6 +1589,7 @@
 Name[ko]=연결 오류
 Name[lt]=Ryšio klaida
 Name[lv]=Savienojuma kļūda
+Name[mai]=संबंधनमे त्रुटि
 Name[ml]=ബന്ധപ്പെടുന്നതില്‍ പിശകു്
 Name[nb]=Tilkoblingsfeil
 Name[nds]=Verbinnen-Fehler
Index: kopete/kopete/kopete.desktop
===================================================================
--- kopete/kopete/kopete.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/kopete.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -40,6 +40,7 @@
 GenericName[ko]=인스턴트 메신저
 GenericName[lt]=Momentinių žinučių klientas
 GenericName[lv]=Tūlītējā ziņojumapmaiņa
+GenericName[mai]=इंसटैंट मैसेंजर
 GenericName[mk]=Инстант гласник
 GenericName[ml]=അതിവേഗ ദൂതന്‍
 GenericName[nb]=Hurtigmelding
@@ -99,6 +100,7 @@
 Comment[ko]=인스턴트 메신저
 Comment[lt]=Momentinių žinučių klientas
 Comment[lv]=Tūlītējas ziņojumapmaiņas programma
+Comment[mai]=इंसटैंट मैसेंजर
 Comment[mk]=Инстант гласник
 Comment[ml]=അതിവേഗ ദൂതന്‍
 Comment[nb]=Hurtigmeldingssystem
Index: kopete/kopete/config/accounts/kopete_accountconfig.desktop
===================================================================
--- kopete/kopete/config/accounts/kopete_accountconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/accounts/kopete_accountconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -42,6 +42,7 @@
 Name[ko]=계정
 Name[lt]=Paskyros
 Name[lv]=Konti
+Name[mai]=खाता
 Name[mk]=Сметки
 Name[ml]=എക്കൌണ്ടുകള്‍
 Name[nb]=Kontoer
Index: kopete/kopete/config/behavior/kopete_behaviorconfig.desktop
===================================================================
--- kopete/kopete/config/behavior/kopete_behaviorconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/behavior/kopete_behaviorconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -43,6 +43,7 @@
 Name[ko]=행동
 Name[lt]=Elgsena
 Name[lv]=Izturēšanās
+Name[mai]=व्यवहार
 Name[mk]=Однесување
 Name[ml]=പെരുമാറ്റം
 Name[nb]=Oppførsel
Index: kopete/kopete/config/plugins/kopete_pluginconfig.desktop
===================================================================
--- kopete/kopete/config/plugins/kopete_pluginconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/plugins/kopete_pluginconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -36,6 +36,7 @@
 Name[ko]=플러그인
 Name[lt]=Įskiepiai
 Name[lv]=Spraudņi
+Name[mai]=प्लगिन
 Name[ml]=സംയോജകങ്ങള്‍
 Name[nb]=Programtillegg
 Name[nds]=Modulen
Index: kopete/kopete/config/status/kopete_statusconfig.desktop
===================================================================
--- kopete/kopete/config/status/kopete_statusconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/status/kopete_statusconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -26,6 +26,7 @@
 Name[ko]=상태
 Name[lt]=Būsena
 Name[lv]=Statuss
+Name[mai]=स्थिति
 Name[ml]=സ്ഥിതി
 Name[pa]=ਹਾਲਤ
 Name[pt]=Estado
Index: kopete/kopete/config/avdevice/kopete_avdeviceconfig.desktop
===================================================================
--- kopete/kopete/config/avdevice/kopete_avdeviceconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/avdevice/kopete_avdeviceconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -22,6 +22,7 @@
 Name[ja]=映像
 Name[km]=វីដេអូ
 Name[ko]=비디오
+Name[mai]=वीडियो
 Name[ml]=ചലനചിത്രം
 Name[oc]=Vidèo
 Name[pa]=ਵੀਡਿਓ
Index: kopete/kopete/config/chatwindow/kopete_chatwindowconfig.desktop
===================================================================
--- kopete/kopete/config/chatwindow/kopete_chatwindowconfig.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/kopete/config/chatwindow/kopete_chatwindowconfig.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -30,6 +30,7 @@
 Name[ko]=대화 창
 Name[lt]=Pokalbių langas
 Name[lv]=Tērzēšanas logs
+Name[mai]=चैट विंडो
 Name[ml]=കുശല ജാലകം
 Name[nb]=Pratevindu
 Name[nds]=Klöönfinster
Index: kopete/libkopete/kopeteaccountmanager.cpp
===================================================================
--- kopete/libkopete/kopeteaccountmanager.cpp	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/libkopete/kopeteaccountmanager.cpp	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -218,6 +218,21 @@
 
 	connect(account, SIGNAL(accountDestroyed(const Kopete::Account *)) , this, SLOT( unregisterAccount(const Kopete::Account *) ));
 
+	if ( !account->identity() )
+	{
+		// the account's Identity must be set here instead of in the Kopete::Account ctor, because there the
+		// identity cannot pick up any state set in the derived Account ctor
+		Identity *identity = Kopete::IdentityManager::self()->findIdentity( account->configGroup()->readEntry("Identity", QString()) );
+		// if the identity was not found, use the default one which will for sure exist
+		// FIXME: get rid of this, the account's identity should always exist at this point
+		if (!identity)
+		{
+			kWarning( 14010 ) << "No identity for account " << account->accountId() << ": falling back to default";
+			identity = Kopete::IdentityManager::self()->defaultIdentity();
+		}
+		account->setIdentity( identity );
+	}
+
 	emit accountRegistered( account );
 	return account;
 }
@@ -394,17 +409,6 @@
 				"Failed to create account for '" << accountId << "'" << endl;
 			continue;
 		}
-		// the account's Identity must be set here instead of in the Kopete::Account ctor, because there the
-		// identity cannot pick up any state set in the derived Account ctor
-		Identity *identity = Kopete::IdentityManager::self()->findIdentity( account->configGroup()->readEntry("Identity", QString()) );
-		// if the identity was not found, use the default one which will for sure exist
-		// FIXME: get rid of this, the account's identity should always exist at this point
-		if (!identity)
-		{
-			kWarning( 14010 ) << "No identity for account " << accountId << ": falling back to default";
-			identity = Kopete::IdentityManager::self()->defaultIdentity();
-		}
-		account->setIdentity( identity );
 	}
 }
 
Index: kopete/libkopete/kopeteversion.h
===================================================================
--- kopete/libkopete/kopeteversion.h	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kopete/libkopete/kopeteversion.h	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -14,10 +14,10 @@
 #ifndef _KOPETE_VERSION_H_
 #define _KOPETE_VERSION_H_
 
-#define KOPETE_VERSION_STRING "0.70.2"
+#define KOPETE_VERSION_STRING "0.70.4"
 #define KOPETE_VERSION_MAJOR 0
 #define KOPETE_VERSION_MINOR 70
-#define KOPETE_VERSION_RELEASE 2
+#define KOPETE_VERSION_RELEASE 4
 #define KOPETE_MAKE_VERSION( a,b,c ) (((a) << 16) | ((b) << 8) | (c))
 
 #define KOPETE_VERSION \
Index: kget/sounds/kget.notifyrc
===================================================================
--- kget/sounds/kget.notifyrc	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kget/sounds/kget.notifyrc	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -507,6 +507,7 @@
 Name[ko]=정보
 Name[lt]=Informacija
 Name[lv]=Informācija
+Name[mai]=सूचना
 Name[ml]=വിവരം
 Name[nb]=Informasjon
 Name[nds]=Informatschonen
Index: filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop
===================================================================
--- filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -111,6 +111,7 @@
 Name[he]=סמבה
 Name[hi]=साम्बा
 Name[hne]=साम्बा
+Name[mai]=साम्बा
 Name[ml]=സാംബ
 Name[ne]=साम्बा
 Name[pa]=ਸਾਂਬਾ
Index: filesharing/advanced/nfs/nfsfile.cpp
===================================================================
--- filesharing/advanced/nfs/nfsfile.cpp	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ filesharing/advanced/nfs/nfsfile.cpp	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -71,7 +71,7 @@
      return 0L;
 
   QString testPath = path.trimmed();
-  if ( testPath[testPath.length()-1] != '/' )
+  if ( !testPath.endsWith('/') )
        testPath += '/';
 
   for (NFSEntry* entry = _entries.first(); entry; entry = _entries.next())
@@ -129,7 +129,7 @@
       completeLine = currentLine;
 
     // is the line continued in the next line ?
-    if ( completeLine[completeLine.length()-1] == '\\' )
+    if ( completeLine.endsWith('\\') )
     {
       continuedLine = true;
       // remove the ending backslash
@@ -144,7 +144,7 @@
     }
 
     // comments
-    if ('#' == completeLine[0]) {
+    if (completeLine.startsWith('#') ) {
       _lines.append(new NFSComment(completeLine));
       continue;
     }
@@ -153,7 +153,7 @@
     QString hosts;
 
     // Handle quotation marks
-    if ( completeLine[0] == '"' ) {
+    if ( completeLine.startsWith('"') ) {
       int i = completeLine.indexOf('"',1);
       if (i == -1) {
         kError() << "NFSFile: Parse error: Missing quotation mark: "
@@ -177,7 +177,7 @@
     }
 
     // normalize path
-    if ( path[path.length()-1] != '/' )
+    if ( !path.endsWith('/') )
             path += '/';
 
     kDebug(5009) << "KNFSShare: Found path: '" << path << "'";
Index: kppp/DB/Provider/Norway/.directory
===================================================================
--- kppp/DB/Provider/Norway/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Norway/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -25,6 +25,7 @@
 Name[ko]=노르웨이
 Name[lt]=Norvegija
 Name[lv]=Norvēģija
+Name[mai]=नार्वे
 Name[ml]=നോര്‍വ്വെ
 Name[nb]=Norge
 Name[nds]=Norwegen
Index: kppp/DB/Provider/Slovenia/.directory
===================================================================
--- kppp/DB/Provider/Slovenia/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Slovenia/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -23,6 +23,7 @@
 Name[ko]=슬로베니아
 Name[lt]=Slovėnija
 Name[lv]=Slovēnija
+Name[mai]=स्लोवेनिया
 Name[ml]=സ്ലോവേനിയ
 Name[nds]=Slowenien
 Name[ne]=स्लोभेनिया
Index: kppp/DB/Provider/Denmark/.directory
===================================================================
--- kppp/DB/Provider/Denmark/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Denmark/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -26,6 +26,7 @@
 Name[ko]=덴마크
 Name[lt]=Danija
 Name[lv]=Dānija
+Name[mai]=डेनमार्क
 Name[ml]=ഡെന്‍മാര്‍ക്ക്
 Name[nb]=Danmark
 Name[nds]=Dänmark
Index: kppp/DB/Provider/Yugoslavia/.directory
===================================================================
--- kppp/DB/Provider/Yugoslavia/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Yugoslavia/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -23,6 +23,7 @@
 Name[ko]=유고슬라비아
 Name[lt]=Jugoslavija
 Name[lv]=Dienvidslāvija
+Name[mai]=युगोस्लाविया
 Name[ml]=യൂഗോസ്ലാവ്യ
 Name[nb]=Serbia og Montenegro
 Name[nds]=Jugoslawien
Index: kppp/DB/Provider/Taiwan/.directory
===================================================================
--- kppp/DB/Provider/Taiwan/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Taiwan/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -16,6 +16,7 @@
 Name[ko]=대만
 Name[lt]=Taivanis
 Name[lv]=Taivāna
+Name[mai]=ताइवान
 Name[ml]=തായ്‌വാന്‍
 Name[ne]=ताइवान
 Name[pa]=ਤਾਈਵਾਨ
Index: kppp/DB/Provider/Sweden/.directory
===================================================================
--- kppp/DB/Provider/Sweden/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Sweden/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -26,6 +26,7 @@
 Name[ku]=Swêd
 Name[lt]=Švedija
 Name[lv]=Zviedrija
+Name[mai]=स्वीडन
 Name[ml]=സ്വീഡന്‍
 Name[nb]=Sverige
 Name[ne]=स्वीडेन
Index: kppp/DB/Provider/New_Zealand/.directory
===================================================================
--- kppp/DB/Provider/New_Zealand/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/New_Zealand/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -24,6 +24,7 @@
 Name[ko]=뉴질랜드
 Name[lt]=Naujoji Zelandija
 Name[lv]=Jaunzēlande
+Name[mai]=न्यूजीलैंड
 Name[ml]=ന്യൂ‌ സിലാന്‍ഡ്
 Name[nds]=Niegseeland
 Name[ne]=न्यूजिल्यान्ड
Index: kppp/DB/Provider/Belarus/.directory
===================================================================
--- kppp/DB/Provider/Belarus/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Belarus/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -25,6 +25,7 @@
 Name[ko]=벨라루스
 Name[lt]=Baltarusija
 Name[lv]=Baltkrievija
+Name[mai]=बेलारूस
 Name[ml]=ബെലാറസ്
 Name[nb]=Hviterussland
 Name[nds]=Wittrussland
Index: kppp/DB/Provider/Portugal/.directory
===================================================================
--- kppp/DB/Provider/Portugal/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Portugal/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -16,6 +16,7 @@
 Name[ko]=포르투갈
 Name[lt]=Portugalija
 Name[lv]=Portugāle
+Name[mai]=पुर्तगाल
 Name[ml]=പോര്‍ച്ചുഗല്‍
 Name[ne]=पोर्चुगल
 Name[pa]=ਪੁਰਤਗਾਲ
Index: kppp/DB/Provider/Ukraine/.directory
===================================================================
--- kppp/DB/Provider/Ukraine/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Ukraine/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -22,6 +22,7 @@
 Name[ko]=우크라이나
 Name[lt]=Ukraina
 Name[lv]=Ukraina
+Name[mai]=उक्रेन
 Name[ml]=ഉക്രൈന്‍
 Name[nb]=Ukraina
 Name[ne]=युक्रेन
Index: kppp/DB/Provider/France/.directory
===================================================================
--- kppp/DB/Provider/France/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/France/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -26,6 +26,7 @@
 Name[ku]=Fransa
 Name[lt]=Prancūzija
 Name[lv]=Francija
+Name[mai]=फ्रांस
 Name[ml]=ഫ്രാന്‍സ്
 Name[nb]=Frankrike
 Name[nds]=Frankriek
Index: kppp/DB/Provider/United_Kingdom/.directory
===================================================================
--- kppp/DB/Provider/United_Kingdom/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/United_Kingdom/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -25,6 +25,7 @@
 Name[ko]=영국
 Name[lt]=Jungtinė Karalystė
 Name[lv]=Lielbritānija
+Name[mai]=यूनाइटेड किंगडम
 Name[ml]=യുണൈറ്റഡ് കിങ്ഡം
 Name[nb]=Storbritannia
 Name[nds]=Grootbritannien
Index: kppp/DB/Provider/Germany/.directory
===================================================================
--- kppp/DB/Provider/Germany/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Germany/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -27,6 +27,7 @@
 Name[ku]=Almanya
 Name[lt]=Vokietija
 Name[lv]=Vācija
+Name[mai]=जर्मनी
 Name[ml]=ജര്‍മ്മനി
 Name[nb]=Tyskland
 Name[nds]=Düütschland
Index: kppp/DB/Provider/Switzerland/.directory
===================================================================
--- kppp/DB/Provider/Switzerland/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Switzerland/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -25,6 +25,7 @@
 Name[ko]=스위스
 Name[lt]=Šveicarija
 Name[lv]=Šveice
+Name[mai]=स्विट्जरलैंड
 Name[ml]=സ്വിറ്റ്സര്‍ലാന്‍ഡ്
 Name[nb]=Sveits
 Name[nds]=Swiez
Index: kppp/DB/Provider/Netherlands/.directory
===================================================================
--- kppp/DB/Provider/Netherlands/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Netherlands/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -25,6 +25,7 @@
 Name[ko]=네덜란드
 Name[lt]=Olandija
 Name[lv]=Nīderlande
+Name[mai]=नीदरलैंड्स
 Name[ml]=നെതര്‍ലാന്റ്സ്
 Name[nb]=Nederland
 Name[nds]=Nedderlannen
Index: kppp/DB/Provider/Austria/.directory
===================================================================
--- kppp/DB/Provider/Austria/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Austria/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -22,6 +22,7 @@
 Name[ko]=오스트리아
 Name[lt]=Austrija
 Name[lv]=Austrija
+Name[mai]=आस्ट्रिया
 Name[ml]=ആസ്ത്രിയ
 Name[nb]=Østerrike
 Name[nds]=Österriek
Index: kppp/DB/Provider/Ireland/.directory
===================================================================
--- kppp/DB/Provider/Ireland/.directory	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ kppp/DB/Provider/Ireland/.directory	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -26,6 +26,7 @@
 Name[ko]=아일랜드
 Name[lt]=Airija
 Name[lv]=Īrija
+Name[mai]=आयरलैंड
 Name[ml]=അയര്‍ലണ്ട്
 Name[nb]=Irland
 Name[nds]=Irland
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.2.3/kdenetwork)	(wersja 974232)
+++ CMakeLists.txt	(.../branches/KDE/4.2/kdenetwork)	(wersja 974232)
@@ -35,8 +35,7 @@
 find_package(Speex)
 find_package(QCA2)
 find_package(JPEG)
-find_package(X11VidMode)
-find_package(Decibel)
+# find_package(Decibel) not used at this time
 find_package(IDN)
 find_package(LibVNCServer)
 

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


